### PR TITLE
Removing the need of bootstrap.sh. Instead, using docker-compose

### LINKED ISSUE #199: Provision the Docker container via Docker file
Now when we have an empty up and running docker container, we should start provision it with all the needed code + dependencies to run the web app. 
Currently, the bootstrap.sh does that.
Make use of Docker file in order to build an image with all that.

UPDATE: this should all work only on DEV env (Vagrant VM) at this stage.

### PR DESCRIPTION
**Pull request by IlanZuckerman**
fix #199 

There is no need in bootstrap.sh anymore.
All of the provision flow is done with a help of docker-compose and Docker file. 

The following was removed, as we agreed in the past that if we are not using it - removing it:
- Mysql and its dependencies installation, including running the mysqk server and building a basic table.
- Pytest test for Mysql

All three envs (vagrant dev, stage and prod) were tested with this PR. All are working, and the app is deployed and running from within a container.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -23,9 +23,10 @@ jobs:
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         flake8 . --count --max-complexity=10 --max-line-length=127 --statistics
 
-    - name: Execute bootstrap.sh
+    - name: Install basic packages
       run: |
-        sudo ./bootstrap.sh
+        sudo apt-get update -y
+        sudo apt-get install -y python3-pip python3-setuptools libsystemd-dev
```
Comment (lioramilbaum) [inline .github/workflows/main.yml:29]: Which process is dependent on those packages?

File: Dockerfile [hunk 2]
```diff
@@ -3,3 +3,11 @@ RUN apt-get update -y && apt-get install -y \
 python3-pip \
 python3-setuptools \
 libsystemd-dev
+
+WORKDIR /code
+COPY . .
```
Comment (lioramilbaum) [inline Dockerfile:8]: This command copies all files from the repo, including the .git folder (if I am not mistaken). Do we have a dockerignore file?

File: docker-compose.yml [hunk 3]
```diff
@@ -1,6 +1,7 @@
-version: '3.9'
+version: '3'
```
Comment (lioramilbaum) [inline docker-compose.yml:1]: Was there a reason to demote the docker compose file version?

File: tests/test_sanity.py [hunk 4]
```diff
@@ -12,6 +13,7 @@ def test_webserver_response_code():
     assert response.status_code == 200
 
 
+@pytest.mark.skip(reason="We dont have mysql server just yet")
```
Comment (lioramilbaum) [inline tests/test_sanity.py:16]: Remove this test all together. We do not need code which is not doing something relevant.

**Review by lioramilbaum — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (IlanZuckerman) [reply .github/workflows/main.yml:29]: > Which process is dependent on those packages?

Libraries from tests/requirements-test.txt require those packages. 
Their installation wouldnt work without those packages. I checked. 
I think that pystemd (which is used i n the tests) is one such library.

File: Dockerfile (hunk 2 above)
Comment (IlanZuckerman) [reply Dockerfile:8]: Adding .dockerignore file.

File: docker-compose.yml (hunk 3 above)
Comment (IlanZuckerman) [reply docker-compose.yml:1]: Yes, it wouldnt run on my Vagrant DEV env for some reason.

File: tests/test_sanity.py (hunk 4 above)
Comment (IlanZuckerman) [reply tests/test_sanity.py:16]: Done

File: .dockerignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
+.git
```
Comment (lioramilbaum) [inline .dockerignore:1]: You might want to take a more general approach. Does ".*" can do the work?

File: .github/workflows/main.yml (hunk 1 above)
Comment (lioramilbaum) [reply .github/workflows/main.yml:29]: Please add a comment that those are required for testing and maybe place this task near the testing task.

File: docker-compose.yml (hunk 3 above)
Comment (lioramilbaum) [reply docker-compose.yml:1]: Please investigate it.

**Review by lioramilbaum — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (IlanZuckerman) [reply .github/workflows/main.yml:29]: I made the 'name' section more informative. I dont this we should move this whole section down near tests execution, it should remain here along with other installations.

File: docker-compose.yml (hunk 3 above)
Comment (IlanZuckerman) [reply docker-compose.yml:1]: Investigated.
I needed to explicitly  specify the desired docker-compose version in the Vagrant file.
So according [official docker doc](https://docs.docker.com/compose/release-notes/) , the latest docker-compose version is 1.28.2
I added this value to Vagrant where the provisioning of the docker-compose happens:
compose_version: "1.28.2"

And so we finish up with the following versions on the VM:
```
ubutu@ip-172-31-27-129:~$ docker --version
Dockner version 20.10.3, build 48d30b5
ubuntu@ip-172-31-27-129:~$ docker-compose --version
docker-compose version 1.28.2, build 67630359
```
Which allows us to use the latest compose file format '3.9'. [This doc](https://docs.docker.com/compose/compose-file/compose-versioning/) specifies the compatibility versions

File: .dockerignore (hunk 5 above)
Comment (IlanZuckerman) [reply .dockerignore:1]: Done. ".*" does the job.
I check that inside the docker container we dont have any of those hidden files or folders under /code path

**Comment by lioramilbaum** [comment]
@IlanZuckerman I don't see the bootstrap.sh file deletion in the PR.

**Comment by IlanZuckerman** [comment]
> @IlanZuckerman I don't see the bootstrap.sh file deletion in the PR.

Removed it and checked that all envs are working (dev, stage)

**Review by lioramilbaum — APPROVED** [review]

**Pull request merged**